### PR TITLE
Make cache_helper report agnostic.

### PR DESCRIPTION
**Pull request by boehmseb**
This allows to have databases that need more than one report file per entry or rely on some other data source.

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- varats/data/databases/blame_diff_metrics_database.py  5
         

Clones added
============
- varats/data/databases/commit_interaction_database.py  1
- varats/data/databases/blame_interaction_degree_database.py  2
- varats/data/databases/blame_interaction_database.py  1
         

Clones removed
==============
+ varats/data/databases/blame_diff_metrics_database.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5669314&bid=18558680)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 11
           

Complexity increasing per file
==============================
- tests/data/test_cache_helper.py  11
- varats/data/databases/blame_diff_metrics_database.py  5
         

Clones added
============
- varats/data/databases/commit_interaction_database.py  1
- varats/data/databases/blame_interaction_database.py  1
- varats/data/databases/blame_interaction_degree_database.py  2
         

Clones removed
==============
+ varats/data/databases/blame_diff_metrics_database.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5669314&bid=18558680)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Complexity increasing per file
==============================
- varats/data/databases/blame_diff_metrics_database.py  5
- tests/data/test_cache_helper.py  11
         

Clones added
============
- varats/data/databases/blame_interaction_database.py  1
- varats/data/databases/commit_interaction_database.py  1
- varats/data/databases/blame_interaction_degree_database.py  2
         

Clones removed
==============
+ varats/data/databases/blame_diff_metrics_database.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5669314&bid=18558680)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- varats/data/databases/blame_diff_metrics_database.py  5
- tests/data/test_cache_helper.py  2
         

Clones added
============
- varats/data/databases/blame_interaction_degree_database.py  2
- varats/data/databases/blame_interaction_database.py  1
- varats/data/databases/commit_interaction_database.py  1
         

Clones removed
==============
+ varats/data/databases/blame_diff_metrics_database.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5669314&bid=18558680)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- tests/data/test_cache_helper.py  2
- varats/data/databases/blame_diff_metrics_database.py  5
         

Clones added
============
- varats/data/databases/blame_interaction_degree_database.py  2
- varats/data/databases/commit_interaction_database.py  1
- varats/data/databases/blame_interaction_database.py  1
         

Clones removed
==============
+ varats/data/databases/blame_diff_metrics_database.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5669314&bid=18558680)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- tests/data/test_cache_helper.py  2
- varats/data/databases/blame_diff_metrics_database.py  5
         

Clones added
============
- varats/data/databases/blame_interaction_database.py  1
- varats/data/databases/commit_interaction_database.py  1
- varats/data/databases/blame_interaction_degree_database.py  2
         

Clones removed
==============
+ varats/data/databases/blame_diff_metrics_database.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5669314&bid=18558680)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- varats/data/databases/blame_diff_metrics_database.py  5
- tests/data/test_cache_helper.py  2
         

Clones added
============
- varats/data/databases/blame_interaction_database.py  1
- varats/data/databases/commit_interaction_database.py  1
- varats/data/databases/blame_interaction_degree_database.py  2
         

Clones removed
==============
+ varats/data/databases/blame_diff_metrics_database.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/se-passau/VaRA-Tool-Suite/pullRequest?prid=5669314&bid=18558680)

**Review by vulder — APPROVED** [review]
lgtm

File: tests/data/test_cache_helper.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+"""Test the cache_helper module."""
+import unittest
+
+import pandas as pd
+
+from tests.test_utils import replace_config
+from varats.data.cache_helper import build_cached_report_table
+
+
+class TestDiffCorrelationOverviewTable(unittest.TestCase):
```
Comment (boehmseb) [inline tests/data/test_cache_helper.py]:
```suggestion
class TestCacheHelper(unittest.TestCase):
```

**Pull request merged**